spring-native-configuration/src/test/java/org/springframework/nativex/util/TestTypeSystem.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright 2021 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.nativex.util;
+
+import java.io.File;
+import java.util.Collections;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Set;
+
+import org.springframework.nativex.type.MissingTypeException;
+import org.springframework.nativex.type.Type;
+import org.springframework.nativex.type.TypeName;
+import org.springframework.nativex.type.TypeSystem;
+
+/**
+ * A type {@link TypeSystem} reading {@literal ./target/classes} able to resolve a given {@link Class} to its {@link Type}.
+ * Should be static to avoid IO overhead.
+ * Allows to {@link #excludePackages(String...)} based on their name - so make sure to {@link #clearExclusions()} during test setup.
+ *
+ * @author Christoph Strobl
+ */
+public class TestTypeSystem extends TypeSystem {
+
+	private Set<String> excludedPackages;
+
+	public TestTypeSystem() {
+		this(Collections.singletonList(new File("./target/classes").toString()));
```

Comment:
Just checking, I had a problem the other day because I had "./target/classes" when it should have been "./target/test-classes" (or both depends on what I really wanted to resolve) - but I guess this is working for you.

Reply:
in this case it worked for me, but I get an error form `TypeSystem.scan` when the a project does not have tests. In that case `target/test-classes` not present and I run into the `IllegalStateException`. Happened when testing `samples/data-rest` with a `NativeConfiguration` that tries to lookup types with specific annotations via the `TypeSystem`.